### PR TITLE
deps: Update dependencies

### PR DESCRIPTION
**Pull request by plakhtiin**
Update all direct, dev-, and transitive dependencies

### CODE REVIEW

File: firebase-rules-coverage/package.json [hunk 1]
```diff
@@ -28,17 +28,18 @@
...
-  }
-}
+    "@jest/globals": "^29.7.0",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^15.14.9",
+    "gts": "^5.3.0",
+    "jest": "^29.7.0",
+    "ts-jest": "^29.1.4",
+    "ts-node": "^10.9.2",
+    "typescript": "~5.1.6"
```
Comment (Prusdrum) [inline firebase-rules-coverage/package.json:42]: Would it make sense to bump the typescript version? `5.4.5` is the latest

File: firebase-rules-generator/package.json [hunk 2]
```diff
@@ -27,17 +27,18 @@
...
-  }
-}
+    "@jest/globals": "^29.7.0",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^15.14.9",
+    "gts": "^5.3.0",
+    "jest": "^29.7.0",
+    "ts-jest": "^29.1.4",
+    "ts-node": "^10.9.2",
+    "typescript": "~5.1.6"
```
Comment (Prusdrum) [inline firebase-rules-generator/package.json:41]: Would it make sense to bump the typescript version? `5.4.5` is the latest

File: firebase-rules-coverage/package.json [hunk 3]
```diff
@@ -28,17 +28,18 @@
...
-    "gts": "^3.1.0",
-    "jest": "^29.6.1",
-    "ts-jest": "^29.1.1",
-    "ts-node": "^10.0.0",
-    "typescript": "^5.1.6"
-  }
-}
+    "@jest/globals": "^29.7.0",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^15.14.9",
```
Comment (Prusdrum) [inline firebase-rules-coverage/package.json:37]: maybe bump to 16 to match the engines field?

File: firebase-rules-generator/package.json [hunk 4]
```diff
@@ -27,17 +27,18 @@
...
-    "gts": "^3.1.0",
-    "jest": "^29.6.1",
-    "ts-jest": "^29.1.1",
-    "ts-node": "^10.0.0",
-    "typescript": "^5.1.6"
-  }
-}
+    "@jest/globals": "^29.7.0",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^15.14.9",
```
Comment (Prusdrum) [inline firebase-rules-generator/package.json:36]: maybe bump to 16 to match the engines field?

File: firebase-rules-coverage/package.json (hunk 1 above)
Comment (plakhtiin) [reply firebase-rules-coverage/package.json:42]: I wanted, but then all tests were crushed. I tried to figure out which dependencies produced it and via empirical method found out that only with ts version "5.1.x" tests are running well.

File: firebase-rules-generator/package.json (hunk 2 above)
Comment (plakhtiin) [reply firebase-rules-generator/package.json:41]: ^^ 
I wanted, but then all tests were crushed. I tried to figure out which dependencies produced it and via empirical method found out that only with ts version "5.1.x" tests are running well.

Comment (Prusdrum) [reply firebase-rules-generator/package.json:38]: Yeah I found it starts to appear in 5.2+
https://devblogs.microsoft.com/typescript/announcing-typescript-5-2/#module-and-moduleresolution-must-match-under-recent-node-js-settings

Comment (plakhtiin) [reply firebase-rules-generator/package.json:38]: The question is: Should we update the resolution version to node 18 as a lowest version for this packages? 🤔

Comment (Prusdrum) [reply firebase-rules-generator/package.json:38]: I tried to do this but it didn't help unfortunately

**Review by Prusdrum — APPROVED** [review]

**Comment by plakhtiin** [comment]
@IchordeDionysos if you will have chance to check and approve, I'd be appreciate ❤️

**Review by HenningKuehl — APPROVED** [review]
RSLGTM

**Comment by plakhtiin** [comment]
@IchordeDionysos this PR is required your review.

**Pull request merged**